packages/graphql/src/translate/where/property-operations/create-connection-operation.ts

```diff
@@ -92,18 +91,51 @@ export function createConnectionOperation({
         if (listPredicateStr === "any" && !connectionField.relationship.typeMeta.array) {
             listPredicateStr = "single";
         }
-        const subquery = new Cypher.RawCypher((env: Cypher.Environment) => {
-            const patternStr = matchPattern.getCypher(env);
-            const whereStr = whereOperator ? whereOperator.getCypher(env) : "";
-            const clause = listPredicateToSizeFunction(listPredicateStr, patternStr, whereStr);
-            return [clause, {}];
-        });
 
-        subqueries = Cypher.concat(subqueries, preComputedSubqueries);
-        operations.push(subquery);
+        const matchClause = new Cypher.Match(matchPattern);
+        const countRef = new Cypher.Variable();
+
+        let whereClause: Cypher.Match | Cypher.With = matchClause;
+        let innerSubqueriesAndWhereClause: Cypher.CompositeClause | undefined;
+
+        if (preComputedSubqueries && !preComputedSubqueries.empty) {
+            whereClause = new Cypher.With("*");
+            innerSubqueriesAndWhereClause = Cypher.concat(preComputedSubqueries, whereClause);
+        }
+
+        if (whereOperator) {
+            const newWhereOperator = listPredicateStr === "all" ? Cypher.not(whereOperator) : whereOperator;
```

Comment:
If there is no where condition an `ALL` filter will result in no elements being returned:

```graphql
{
    movies(where: { genresConnection_ALL: {} }) {
        title
    }
}
```

results in

```
MATCH (this:`Movie`)
CALL {
    WITH this
    MATCH (this)-[this0:IN_GENRE]->(this1:`Genre`)
    RETURN count(this0) AS var2
}
WITH *
WHERE var2 = 0
RETURN this { .title } AS this
```

is this intended?

Reply:
This was not previously supported and before this PR resulted in invalid cypher. I've created a feature request so the team can consider if we want to support this